metal/drivers/sifive_hca1_regs.h

```diff
@@ -0,0 +1,835 @@
+            uint32_t   _reserved13[12U];
+    __IM    uint32_t   SHA_REV;           /**< Offset: 0x170 (R/ ) SHA Revision */
+            uint32_t   _reserved14;
+    __IOM   uint32_t   SHA_CR;            /**< Offset: 0x178 (R/W) SHA control */
+            uint32_t   _reserved15;
+    __IM    uint64_t   HASH[8U];          /**< Offset: 0x180 (R/ ) HASH result from SHA */
+            uint32_t   _reserved16[32U];
+    __IM    uint32_t   PKA_REV;           /**< Offset: 0x240 (R/ ) PKA Revision */
+            uint32_t   _reserved17;
+    __IOM   uint32_t   PKA_CR;            /**< Offset: 0x248 (R/W) PKA control cegister */
+            uint32_t   _reserved18;
+    __IM    uint32_t   PKA_SR;            /**< Offset: 0x250 (R/ ) PKA control */
+            uint32_t   _reserved19;
+    __IOM   uint32_t   PKA_OPA;           /**< Offset: 0x258 (R/W) PKA Operand A Address */
+            uint32_t   _reserved20[3U];
+    __IOM   uint32_t   PKA_OPB;           /**< Offset: 0x268 (R/W) PKA Operand B Address */
+            uint32_t   _reserved21[3U];
+    __IOM   uint32_t   PKA_RES;           /**< Offset: 0x278 (R/W) PKA Result Address */
+            uint32_t   _reserved22[3U];
+    __IOM   uint32_t   PKA_MOD;           /**< Offset: 0x288 (R/W) PKA Modulo Address */
+            uint32_t   _reserved23[13U];
+    __IM    uint32_t   TRNG_REV;          /**< Offset: 0x2C0 (R/ ) TRNG Revision */
+            uint32_t   _reserved24;
+    __IOM   uint32_t   TRNG_CR;           /**< Offset: 0x2C8 (R/W) TRNG control */
+            uint32_t   _reserved25;
+    __IOM   uint32_t   TRNG_SR;           /**< Offset: 0x2D0 (R/W) TRNG status */
+            uint32_t   _reserved26;
+    __IM    uint32_t   TRNG_DATA;         /**< Offset: 0x2D8 (R/ ) TRNG data */
+            uint32_t   _reserved27;
+    __IOM   uint32_t   TRNG_TRIM;         /**< Offset: 0x2E0 (R/W) TRNG trim */
+            uint32_t   _reserved28;
+    __IM    uint32_t   TRNG_OSC1_CNT;     /**< Offset: 0x2E8 (R/ ) TRNG osc1 counter */
+            uint32_t   _reserved29;
+    __IM    uint32_t   TRNG_OSC2_CNT;     /**< Offset: 0x2F0 (R/ ) TRNG osc1 counter */
+} HCA_Type;
+
+/**
+ * Structure type to access HCA Revision (HCA_REV)
+ */
+typedef union _HCA_HCA_REV {
+    struct {
+        uint32_t PATCHREV:4;       /**< bit:   0..3  HCA Patch Revision */
+        uint32_t MINORREV:4;       /**< bit:   4..7  HCA Minor Revision */
+        uint32_t MAJORREV:4;       /**< bit:  8..11  HCA Major Revision */
+        uint32_t _reserved0:4;     /**< bit: 12..15  (reserved) */
+        uint32_t CONFIG:5;         /**< bit: 16..20  HCA Configuration */
+        uint32_t _reserved1:11;    /**< bit: 21..31  (reserved) */
+    } b;                           /**< Structure used for bit access */
+    uint32_t w;                    /**< Structure used for word access */
+} HCA_HCA_REV_Type;
+
+/* HCA HCA Revision: Patch Revision */
+#define HCA_HCA_REV_PATCHREV_Pos        0U
```

Comment:
You should follow the current coding styles, in that all macro/#define should be all caps.
We should also uses the full word and not abbreviation, for POSITION and MASK.

Reply:
This file is CMSIS like, so I follow the coding style of CMSIS.